workflow/workflow_test.go

```diff
@@ -1181,11 +1181,13 @@ steps:
     step: wait
     deploy:
       deployer_name: "test-impl"
-      # Set the deploy time long enough for there to not be a race between this exiting and the cancellation.
-      # There is no dependency on stop_if, so we need to enforce order with this.
+      # stop_if doesn't create any dependency, so we must keep the step in the deployment
+      # stage long enough for the cancellation to occur. Otherwise there would be a race.
       deploy_time: 50 # ms
     input:
-      wait_time_ms: 20
+      # The actual wait time should not matter for this test because the intention is to
+      # to cancel it before it is run.
+      wait_time_ms: 0
```

Comment:
The second comment is excellent.  That's what we should say in the first comment:  the intention is for the cancellation to occur during the deployment, so we configure the deployment to take long enough to ensure that that happens.  It's fine to point out that the `stop_if` will run immediately, because it has no dependency, but I don't think it helps to mention that there would otherwise be a race.  Let's stick to what we are trying to produce rather than talking about what we're trying to avoid.